src/ui/common/src/components/integrations/dialogs/deleteIntegrationDialog.tsx

```diff
@@ -64,7 +93,35 @@ const DeleteIntegrationDialog: React.FC<Props> = ({
     return state.integrationReducer.operators;
   });
 
-  if (
+  const isStorage = config.use_as_storage === 'true';
+  let isCurrentStorage = isStorage;
+  if (isStorage && metadataStorageConfig) {
+    const storageConfig = convertIntegrationConfigToMetadataStorageConfig(
+      config,
+      metadataStorageConfig,
+      integrationType
+    );
+    // Check deep equality
+    isCurrentStorage = isEqual(storageConfig, metadataStorageConfig);
```

Comment:
Could you write a small function to check if the storageConfig and metadataStorageConfig are equal?

This will save us from having to bring in lodash as a dependency.

Reply:
Done!